buildSrc/src/main/kotlin/partiql.versions.kt

```diff
@@ -45,10 +45,10 @@ object Versions {
     const val kotlinxCollections = "0.3.5"
     const val picoCli = "4.7.0"
     const val kasechange = "1.3.0"
-    const val ktlint = "11.6.0"
     const val pig = "0.6.2"
     const val kotlinxCoroutines = "1.6.0"
     const val kotlinxCoroutinesJdk8 = "1.6.0"
+    const val ktlint = "0.42.1" // we're on an old version of ktlint. TODO upgrade https://github.com/partiql/partiql-lang-kotlin/issues/1418
```

Comment:
How did we go back from 11.6.0 to 0.42.1? Was the original one for the the gradle plugin, and this is for the custom rule dependencies?

Reply:
There were a couple confusing things here previously.
- The `ktlint` dependency defined previously actually referred to the ktlint gradle plugin -- https://github.com/JLLeitschuh/ktlint-gradle and not the actual `ktlint` Maven dependency -- https://github.com/pinterest/ktlint
- The 11.6.0 dependency (defined in `buildSrc/src/main/kotlin/partiql.versions.kt`) was actually unused. We were actually using the 10.2.1 ktlint gradle plugin defined in `buildSrc/build.gradle.kts`

The v0.42.1 refers to the default `ktlint` version that the 10.2.1 kotlin gradle plugin uses (see [their source](https://github.com/JLLeitschuh/ktlint-gradle/blob/v10.2.1/plugin/gradle/libs.versions.toml#L3)). I chose to use the same `ktlint` version for the custom rules that we were using for the default `ktlintCheck`/`ktlintFormat` runs. I didn't upgrade the default `ktlint` version from v0.42.1 to a newer one since there are some additional rules that our code is not compliant with. We can perform this `ktlint` version upgrade in a future PR (tracked in https://github.com/partiql/partiql-lang-kotlin/issues/1418).

Hope that makes things clearer.